frontend/src/constants/Interfaces.ts

```diff
@@ -0,0 +1,48 @@
+import { headerStates, bottomStates, pageStates, globalStates } from './States';
+import { actionTypes } from './ActionTypes';
+
+export interface Item {
+    id: number,
+    name: string,
+    tags: string[],
+    price?: number, // this should be in cents
+    store?: string
+}
+
+export interface Store {
+    isSingleStore?: boolean,
+    searchQuery?: string,
+    items: Item[]
+}
+
+export interface State {
+    headerState: headerStates,
+    bottomState: bottomStates,
+    pageState: pageStates,
+    store: Store
+}
+
+export interface searchPayload {
+    searchQuery: string
+}
+
+export interface addItemPayload {
+    addedItem: Item
+}
+
+export interface removeItemPayload {
+    removeItem: number // ID of item
+}
+
+export interface updateItemsPayload {
+    updatedItems: Item[]
+}
+
+export interface switchStorePayload {
+    isSingleStore: boolean
+}
+
+export interface Action {
```

Comment:
While what you did here works, what I usually do for my Redux is to have separate interfaces for each type of action. That way, I don't have to make separate interfaces for the payloads later. I think it will also help with some of the type-casting that you had to do in `Reducer.ts`